### PR TITLE
[AEROGEAR-2613] make metrics bindable

### PR DESCRIPTION
**Pull request by JameelB**
## Description
Metrics needs to be bindable and be available within the integration component of the mobile client component.

## Progress
- [x] Set `bindable` to be true
- [x] Add bind playbook
- [x] Add bind role
- [x] Add unbind playbook
- [x] Add unbind role
- [x] Add bind parameters
- [x] Add mobile-client-enabled tag

## Verification Steps

- [ ] Provision a mobile client.
- [ ] Ensure that metrics is available as an option within the integration section of the mobile client.
- [ ] Provision metrics successfully
- [ ] Integrate metrics with a mobile client.
- [ ] Ensure that a secret and a service binding is created.
- [ ] Ensure that the secret is labelled correctly with `clientId`, `serviceName`, `serviceInstanceId` and `mobile:enabled`
- [ ] Ensure that the correct information is displayed within the integration section of the mobile client.
- [ ] Ensure that the right client config is displayed within the client config component. (Has `id`, `config` and `uri`)
- [ ] Delete the integration
- [ ] Ensure that the secret and the service binding created during binding is removed.
- [ ] Deprovision metrics.
- [ ] Ensure that it has cleaned up properly.

## Screenshots
image

image

## Additional Notes
Related JIRA Ticket - https://issues.jboss.org/browse/AEROGEAR-2613
**Expected client config**
```
    services: [{
      "id": "metrics",
      "name": "metrics",
      "type": "metrics",
      "url": "<app-metrics-url>",
      "config": {}
    }]
```
**Purpose of binding**
Integration with metrics allows the mobile client to send metrics to an Aerogear App Metrics instance.

### CODE REVIEW

File: roles/bind-metrics-apb/templates/bind_secret.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  namespace: "{{ namespace }}"
+  labels:
+    mobile: enabled
+    name: "{{ prometheus_secret_name }}-{{ CLIENT_ID }}"
+    clientId: "{{ CLIENT_ID }}"
+    serviceName: "{{ prometheus_secret_name }}"
+    serviceInstanceId: "{{ _apb_service_instance_id }}"
+stringData:
+  clientId: "{{ CLIENT_ID }}"
+  config: "{}"
```
Comment (dimitraz) [inline roles/bind-metrics-apb/templates/bind_secret.yml]: Is this field used?

File: playbooks/unbind.yml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+- name: metrics-apb playbook to bind the application
```
Comment (dimitraz) [inline playbooks/unbind.yml]: metrics-apb playbook to unbind the application

File: roles/bind-metrics-apb/templates/bind_secret.yml (hunk 1 above)
Comment (JameelB) [reply roles/bind-metrics-apb/templates/bind_secret.yml]: This is empty for now because there is no content required yet for it but is needed for the client config.

Comment (philbrookes) [reply roles/bind-metrics-apb/templates/bind_secret.yml]: didn't we update the UI to default to {} if this field was not present, so that this would not be necessary?

File: apb.yml [hunk 3]
```diff
@@ -1,10 +1,11 @@
 version: 1.0
 name: metrics-apb
 description: Installs a metrics service based on Prometheus and Grafana
-bindable: False
+bindable: True
```
Comment (philbrookes) [inline apb.yml]: @maleck13 considering that metrics find scrape targets via service discovery, when would any service / mobile-client want to bind to metrics? What is the use-case for binding to metrics?

Comment (maleck13) [reply apb.yml]: It is to allow the app to send metrics in

File: roles/bind-metrics-apb/templates/bind_secret.yml [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+kind: Secret
+metadata:
+  name: "{{ prometheus_secret_name }}-{{ CLIENT_ID }}"
+  namespace: "{{ namespace }}"
+  labels:
+    mobile: enabled
+    name: "{{ prometheus_secret_name }}-{{ CLIENT_ID }}"
+    clientId: "{{ CLIENT_ID }}"
+    serviceName: "{{ prometheus_secret_name }}"
+    serviceInstanceId: "{{ _apb_service_instance_id }}"
```
Comment (dimitraz) [inline roles/bind-metrics-apb/templates/bind_secret.yml]: I don't think we have to label the bind secret with `serviceInstanceId`

File: roles/bind-metrics-apb/templates/bind_secret.yml (hunk 1 above)
Comment (JameelB) [reply roles/bind-metrics-apb/templates/bind_secret.yml]: I think @sedroche will be updating this with the UI change for [2413](https://issues.jboss.org/browse/AEROGEAR-2413). Need this field for now for verification

Comment (sedroche) [reply roles/bind-metrics-apb/templates/bind_secret.yml]: This needs to go into the `mobileClientConfig` component. As that is part of an upstream PR can you add this @philbrookes ?

**Comment by david-martin** [comment]
I'd recommend this change gets verified with an app (on emulator or device) that's using the AeroGear SDK so the mobile-services.json that gets generated after this change is still as expected.

@josemigallas @wtrocki Can you recommend steps and an app to use for this (I'm not sure where we are with the showcase app and which repo it's currently maintained in)

**Comment by wtrocki** [comment]
From what I see this is not going to work with current SDK as we expect id to be `metrics`. I will recommend to send update to the list and service teams going to create corresponding SDK tickets to adjust that. 

@david-martin At any point of the time you should be able to get any link to repo for SDK, template etc. information on https://github.com/aerogear/mobile-core README (website in future. If this will be outdated let me know! It is best to maintain this central point of information for community.

**Comment by maleck13** [comment]
@wtrocki after our conversation and mail thread about about the sdk are we happy to merge here now?

**Comment by wtrocki** [comment]
Yes. This is no longer blocking any SDK's. This change should work with IOS and Cordova. Android will be also fixed soon.

TL;DR - Merge it (after build will pass)
